DREAM/Optimization/tasdreamOptimizationTests.cpp

```diff
@@ -0,0 +1,194 @@
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice, this list of conditions
+ *    and the following disclaimer in the documentation and/or other materials provided with the distribution.
+ *
+ * 3. Neither the name of the copyright holder nor the names of its contributors may be used to endorse
+ *    or promote products derived from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+ * INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+ * OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA,
+ * OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ * UT-BATTELLE, LLC AND THE UNITED STATES GOVERNMENT MAKE NO REPRESENTATIONS AND DISCLAIM ALL WARRANTIES, BOTH EXPRESSED AND
+ * IMPLIED. THERE ARE NO EXPRESS OR IMPLIED WARRANTIES OF MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE, OR THAT THE USE OF
+ * THE SOFTWARE WILL NOT INFRINGE ANY PATENT, COPYRIGHT, TRADEMARK, OR OTHER PROPRIETARY RIGHTS, OR THAT THE SOFTWARE WILL
+ * ACCOMPLISH THE INTENDED RESULTS OR THAT THE SOFTWARE OR ITS USE WILL NOT RESULT IN INJURY OR DAMAGE. THE USER ASSUMES
+ * RESPONSIBILITY FOR ALL LIABILITIES, PENALTIES, FINES, CLAIMS, CAUSES OF ACTION, AND COSTS AND EXPENSES, CAUSED BY, RESULTING
+ * FROM OR ARISING OUT OF, IN WHOLE OR IN PART THE USE, STORAGE OR DISPOSAL OF THE SOFTWARE.
+ */
+
+#ifndef __TASMANIAN_TASDREAM_OPTIMIZATION_TESTS_CPP
+#define __TASMANIAN_TASDREAM_OPTIMIZATION_TESTS_CPP
+
+#include "TasmanianOptimization.hpp"
+#include "tasdreamExternalTests.hpp"
+
+namespace TasOptimization {
+
+// Unit tests for TasOptimization::ParticleSwarmState.
+bool testParticleSwarmState(double tol=1e-12) {
+    bool pass = true;
+
+    // Check size of the accessible vectors generated by different constructors.
+    int num_dimensions = 2;
+    int num_particles = 15;
+    std::vector<double> dummy_positions(num_dimensions * num_particles);
+    std::vector<double> dummy_velocities(num_dimensions * num_particles);
+    std::vector<ParticleSwarmState> states = {
+        ParticleSwarmState(num_dimensions, num_particles),
+        ParticleSwarmState(num_dimensions, std::move(dummy_positions), std::move(dummy_velocities))
+    };
+    for (int i=0; i<2; i++) {
+        pass = pass & ((int) states[i].getParticlePositions().size() == num_dimensions * num_particles);
```

Comment:
Use `and` as opposed to`&` (also it should have been `&&`)

Reply:
Done.